lib/vrl/stdlib/src/to_bool.rs

```diff
@@ -173,87 +173,41 @@ impl Expression for ToBoolFn {
-//             def: TypeDef { kind: Kind::Boolean, ..Default::default() },
-//         }
-
-//         string_fallible {
-//             expr: |_| ToBoolFn { value: lit!("foo").boxed() },
-//             def: TypeDef { fallible: true, kind: Kind::Boolean, ..Default::default() },
-//         }
-
-//         map_fallible {
-//             expr: |_| ToBoolFn { value: map!{}.boxed() },
-//             def: TypeDef { fallible: true, kind: Kind::Boolean, ..Default::default() },
-//         }
-
-//         array_fallible {
-//             expr: |_| ToBoolFn { value: array![].boxed() },
-//             def: TypeDef { fallible: true, kind: Kind::Boolean, ..Default::default() },
-//         }
-
-//         timestamp_fallible {
-//             expr: |_| ToBoolFn { value: Literal::from(chrono::Utc::now()).boxed() },
-//             def: TypeDef { fallible: true, kind: Kind::Boolean, ..Default::default() },
-//         }
-
-//         fallible_value_without_default {
-//             expr: |_| ToBoolFn { value: lit!("foo").boxed() },
-//             def: TypeDef {
-//                 fallible: true,
-//                 kind: Kind::Boolean,
-//                 ..Default::default()
-//             },
-//         }
-//     ];
-
-//     #[test]
-//     fn to_bool() {
-//         use crate::map;
+        string_true {
+            args: func_args![value: "true"],
+            want: Ok(true),
+            tdef: TypeDef::new().fallible().boolean(),
+        }
 
-//         let cases = vec![
-//             (
-//                 map!["foo": "true"],
-//                 Ok(Value::Boolean(true)),
-//                 ToBoolFn::new(Box::new(Path::from("foo"))),
-//             ),
-//             (
-//                 map!["foo": 20],
-//                 Ok(Value::Boolean(true)),
-//                 ToBoolFn::new(Box::new(Path::from("foo"))),
-//             ),
-//         ];
+        string_false {
+            args: func_args![value: "no"],
+            want: Ok(false),
+            tdef: TypeDef::new().fallible().boolean(),
+        }
 
-//         let mut state = state::Program::default();
+        string_error {
+            args: func_args![value: "cabbage"],
+            want: Err(r#"Invalid boolean value "cabbage""#),
+            tdef: TypeDef::new().fallible().boolean(),
+        }
 
-//         for (object, exp, func) in cases {
-//             let mut object: Value = object.into();
-//             let got = func
-//                 .resolve(&mut ctx)
-//                 .map_err(|e| format!("{:#}", anyhow::anyhow!(e)));
+        number_true {
+            args: func_args![value: 20],
+            want: Ok(true),
+            tdef: TypeDef::new().infallible().boolean(),
+        }
 
-//             assert_eq!(got, exp);
-//         }
-//     }
-// }
+        number_false {
+            args: func_args![value: 0],
+            want: Ok(false),
+            tdef: TypeDef::new().infallible().boolean(),
+        }
+    ];
```

Comment:
It would probably be good for all `to_*` functions to test all possible permutations. But, we can defer that to a follow-up PR 👍